relay.go

```diff
@@ -654,6 +664,80 @@ func (r *Relayer) handleLocalCallReq(cr *lazyCallReq) bool {
 	return true
 }
 
+func (r *Relayer) fragmentingSend(f *lazyCallReq, relayToDest relayItem, origID uint32) error {
+	fragWriter := newFragmentingWriter(
+		NullLogger, r.newFragmentSender(relayToDest.destination, f, origID),
+		f.checksumType.New(),
+	)
+	defer r.conn.opts.FramePool.Release(f.Frame)
+
+	arg2Writer, err := fragWriter.ArgWriter(f.isArg2Fragmented)
+	if err != nil {
+		return fmt.Errorf("get arg2 writer: %v", err)
+	}
+
+	if err := writeArg2WithAppends(arg2Writer, f.arg2(), f.arg2appends); err != nil {
+		return fmt.Errorf("write arg2: %v", err)
+	}
+	if f.isArg2Fragmented {
+		return nil
+	}
+
+	if err := NewArgWriter(fragWriter.ArgWriter(true)).Write(f.arg3()); err != nil {
+		return errors.New("arg3 write failed")
+	}
+
+	return nil
+}
+
+func writeArg2WithAppends(w io.WriteCloser, arg2 []byte, appends []keyVal) (err error) {
+	defer w.Close()
+
+	var nh uint16
+	if len(arg2) > 0 {
+		nh = binary.BigEndian.Uint16(arg2[:2])
```

Comment:
what if `len(arg2) == 1`? this would panic in that case.

i think if the length of arg2 < 2, we should return an error immediately.

Reply:
there are some cases where len(arg2) == 0 (no arg2), so I think we'll still have to accommodate that. I'll change the check to avoid the panic

Reply:
I'm not sure I understand when `len(arg2) == 0`, since that seems like a protocol error. No arg2 should result in a length prefix of 0 (2 bytes).

Reply:
you're right. I somehow ran into some test failures before where it seemed like it was expecting arg2 to be empty, but apparently it was due to some other bug in prior versions. will drop the check